packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -36,19 +37,94 @@ enum _ScaffoldSlot {
   statusBar,
 }
 
+/// A position that the [FloatingActionButton] can dock to in a [Scaffold].
+/// 
+/// [getOffset] 
+class FabPosition {
```

Comment:
presumably this should be abstract?

Reply:
Oh, I see, you intend this not to be extended but to provide a callback.

Interesting. Can you elaborate on this? Why not just make this a class that gets extended? You do extend it as well, which is odd.

Reply:
I've decided to take the inheritance approach to this class, the field is now an abstract method.